xds/src/test/java/io/grpc/xds/XdsClientImplTestBase.java

```diff
@@ -3640,6 +3766,39 @@ private  BootstrapInfo buildBootStrap(String serverUri) {
         .build();
   }
 
+  private <T extends ResourceUpdate> DiscoveryRpcCall startResourceWatcher(
+      XdsResourceType<T> type, String name, ResourceWatcher<T> watcher, Executor executor) {
+    FakeClock.TaskFilter timeoutTaskFilter;
+    switch (type.typeName()) {
+      case "LDS":
+        timeoutTaskFilter = LDS_RESOURCE_FETCH_TIMEOUT_TASK_FILTER;
+        xdsClient.watchXdsResource(type, name, watcher, executor);
+        break;
+      case "RDS":
+        timeoutTaskFilter = RDS_RESOURCE_FETCH_TIMEOUT_TASK_FILTER;
+        xdsClient.watchXdsResource(type, name, watcher, executor);
+        break;
+      case "CDS":
+        timeoutTaskFilter = CDS_RESOURCE_FETCH_TIMEOUT_TASK_FILTER;
+        xdsClient.watchXdsResource(type, name, watcher, executor);
+        break;
+      case "EDS":
+        timeoutTaskFilter = EDS_RESOURCE_FETCH_TIMEOUT_TASK_FILTER;
+        xdsClient.watchXdsResource(type, name, watcher, executor);
+        break;
+      default:
+        throw new AssertionError("should never be here");
+    }
+
+    DiscoveryRpcCall call = resourceDiscoveryCalls.poll();
+    call.verifyRequest(type, Collections.singletonList(name), "", "", NODE);
```

Comment:
poll could return null if resourceDiscoveryCalls is empty.  It would be better to do an explicit check with an explanatory message than get an NPE.